### PR TITLE
 Fix bug in normalisation neuron

### PR DESCRIPTION
**Pull request by jzstark**
This PR fix the error introduced in my previous one as shown in #225. I'm very sorry for didn't checking the code thoroughly. For future reference, I added links to the [original paper](https://arxiv.org/abs/1502.03167) on Batch Normalisation and the [implementation](https://github.com/keras-team/keras/blob/master/keras/layers/normalization.py#L130) in Keras.

### CODE REVIEW

**Comment by jzstark** [comment]
I just noticed that after commit [9b9ad](https://github.com/owlbarn/owl/commit/9b9ad77b5fd01f7a714f30018e89b056396e1d51) Travis fails with message:
```
# File "src/owl/dense/owl_dense_ndarray_generic.ml", line 3604, characters 11-34:
# Error: Unbound value Owl_utils.Array.replace
```
Once I copy this `replace` function into `owl_dense_ndarray_generic.ml` and call it instead of `Owl_utils.Array.replace`, the Travis test is successful again.  I'm still trying to figure out what may have caused this issue.

**Comment by ryanrhymes** [comment]
Is it because Owl core tries to download Owl_base from OPAM now rather than using the one in the source tree?  @mor1

File: src/base/neural/owl_neural_neuron.ml [hunk 1]
```diff
@@ -2043,14 +2043,16 @@ module Make
...
-      let a = F (1. /. float_of_int (shape x).(l.axis)) in
       if l.training = true then (
+        let a = F (float_of_int ((numel x) / (shape x).(l.axis))) in
         let s = Owl_utils_array.range 0 (Array.length l.in_shape) in
         let s = List.filter (fun x -> x != l.axis) (Array.to_list s)
           |> Array.of_list in
-        let mu' = Maths.(a * (sum_reduce ~axis:s x)) in
-        let var' = Maths.(a * (sum_reduce ~axis:s (x * x))) in
+        let mu' = Maths.((sum_reduce ~axis:s x) / a) in
+        let var' = Maths.((sum_reduce ~axis:s (x * x)) / a) in
```
Comment (ryanrhymes) [inline src/base/neural/owl_neural_neuron.ml]: Can we change division to multiplication, by flipping the numerator and denominator when you calculate a? Because multiplication is cheaper than division, also for backprop phase.

Just suggesting ...

File: src/base/neural/owl_neural_neuron.ml [hunk 2]
```diff
@@ -2043,14 +2043,16 @@ module Make
       mkpri l |> Array.map copy_primal' |> update l';
       l'
 
+    (* Reference: https://arxiv.org/abs/1502.03167.
+       Implementaton in Keras: https://bit.ly/2vBsvgI.*)
     let run x l =
-      let a = F (1. /. float_of_int (shape x).(l.axis)) in
       if l.training = true then (
+        let a = F (float_of_int ((numel x) / (shape x).(l.axis))) in
```
Comment (ryanrhymes) [inline src/base/neural/owl_neural_neuron.ml]: Are you sure this is correct? Shouldn't it be `F ((float_of_int (numel x)) / (float_of_int (shape x).(l.axis)))` ... please double check the reference.

And the link to the original paper is still missing.
